bahmnicore-omod/src/main/java/org/bahmni/module/bahmnicore/web/v1_0/controller/AdminImportController.java

```diff
@@ -119,10 +118,15 @@
     @Qualifier("adminService")
     private AdministrationService administrationService;
 
+    @Autowired
+    private RegistrationPageService registrationPageService;
+
     @RequestMapping(value = baseUrl + "/patient", method = RequestMethod.POST)
     @ResponseBody
-    public boolean upload(@RequestParam(value = "file") MultipartFile file) throws IOException {
+        public boolean upload(@RequestParam(value = "file") MultipartFile file, @RequestHeader("Host") String host, @RequestHeader(value = "Origin", required = false) String origin, @RequestHeader(value = "Referer", required = false) String referer) throws IOException {
         try {
+            registrationPageService.setProtocol(getProtocol(origin, referer));
```

Comment:
@sivareddytw why are we setting the host, origin and referer to registrationPageService ?

Reply:
@binduk as per discussion with Angshu in PAT calls, UI and backend can be deployed in two machines. In that case, fetching registration configuration won't work. So, fetching registration configuration using HTTP protocol.